domain-services/fiscal-event-service/src/main/java/org/egov/validator/FiscalEventValidator.java

```diff
@@ -86,130 +87,87 @@ public void validateFiscalEventPushPost(FiscalEventRequest fiscalEventRequest) {
-                }
-            }
-            if (!parentEventIds.isEmpty() && StringUtils.isNotBlank(tenantId)) {
-                Criteria criteria = new Criteria();
-                List<String> ids = new ArrayList<>(parentEventIds);
-                criteria.setTenantId(tenantId);
-                criteria.setIds(ids);
-
-                List<Object> fiscalEventList = eventRepository.searchFiscalEvent(criteria);
-                if (fiscalEventList == null || fiscalEventList.isEmpty()) {
-                    errorMap.put(MasterDataConstants.PARENT_EVENT_ID, "Linked event id(s) : " + parentEventIds + " doesn't exist in the system.");
-                } else if (!fiscalEventList.isEmpty() && (fiscalEventList.size() != parentEventIds.size())) {
-                    List<FiscalEvent> dbFiscalEvents = mapperUtil.mapDereferencedFiscalEventToFiscalEvent(fiscalEventList);
-                    Set<String> missingParentEventIds = new HashSet<>();
-                    for (String parentEventId : parentEventIds) {
-                        boolean isExist = false;
-                        for (FiscalEvent fiscalEvent : dbFiscalEvents) {
-                            if (fiscalEvent != null && StringUtils.isNotBlank(fiscalEvent.getLinkedEventId())
-                                    && fiscalEvent.getLinkedEventId().equals(parentEventId)) {
-                                isExist = true;
-                                break;
-                            }
-                        }
-                        if (!isExist) {
-                            missingParentEventIds.add(parentEventId);
-                        }
-                    }
-                    if (!missingParentEventIds.isEmpty()) {
-                        errorMap.put(MasterDataConstants.PARENT_EVENT_ID, "Linked event id(s) : " + missingParentEventIds + " doesn't exist in the system.");
-                    }
-                }
-            }
+            errorMap.put(MasterDataConstants.EVENT_TYPE, "Fiscal event type is missing");
+            isMissing = true;
         }
+        // TODO: ADD tenant id not null check
 
     }
 
-    /**
-     * Validate the request header
-     *
-     * @param requestHeader
-     */
-    private void validateReqHeader(RequestHeader requestHeader) {
-        if (requestHeader == null)
-            throw new CustomException("REQUEST_HEADER", "Request header is missing");
-        if (requestHeader.getUserInfo() == null)
-            throw new CustomException("USER_INFO", "User info is missing in request header");
-        if (StringUtils.isBlank(requestHeader.getUserInfo().getUuid()))
-            throw new CustomException("USER_INFO", "User info is missing in request header");
+    private void validateBatchSize(FiscalEventRequest fiscalEventRequest) {
+        if (StringUtils.isNotBlank(fiscalEventConfiguration.getFiscalEventPushReqMaxSize())) {
+            int size = Integer.parseInt(fiscalEventConfiguration.getFiscalEventPushReqMaxSize());
```

Comment:
In `FiscalEventConfiguration`, make this variable an Integer. Spring will automatically convert String to Integer when you declare the variable an Integer. You wouldn't need to parse it here.